### PR TITLE
[WIP] Util.sweep() for sweeping paper wallets

### PR DESCRIPTION
**Pull request by christroutner**
This PR is a work-in-progress (WIP) and should not be merged. It is created to solicit feedback.

## Scope
At @cgcardona's request, I added a sweep function to the BITBOX Utils library. This is a high-level function that can be used to 'sweep' a paper wallet and claim the funds on it.

## Food for Thought
I'm not totally satisfied with how this PR has shaped up. The functionality is in place, and I've added two prototype integration tests to exercise the function. I'll place call-outs in the code, but here are a few things I don't like about this implementation:

- I have to pull a whole bunch of the other libraries into the Util library in order to execute the sweep functionality. This makes me think that this sweep functionality is more appropriate for a different, higher-level library like [bch-cli-wallet](https://github.com/Bitcoin-com/bch-cli-wallet), or should be created as an example in the examples directory.

- Because the new sweep function makes a few calls to rest.bitcoin.com, it's going to be odd to create unit tests for it. End-to-end tests will be required to effectively test it. Unit and integration tests will provide minimal confidence on its expected behavior.

## To-Do
- Discuss the practicality of including this function in BITBOX.
- Add more comprehensive integration tests
- Add mocked unit tests.

### CODE REVIEW

File: lib/Util.ts [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
+// Pull in additional libraries required to support the sweep function.
+import { Address } from "./Address"
+const addressLib = new Address(REST_URL)
+import { ECPair } from "./ECPair"
+const ecPair = new ECPair(addressLib)
+import { TransactionBuilder } from "./TransactionBuilder"
+import { BitcoinCash } from "./BitcoinCash"
+const bitcoinCash = new BitcoinCash(addressLib)
+
 export interface AddressDetails {
```
Comment (christroutner) [inline lib/Util.ts]: These are all libraries that need to be pulled in to perform the sweep functionality. Something about how I pulled these in are causing the unit tests to fail. Even if they pass, all these cross-linked libraries are a red flag, which makes me think this sweep functionality should go into a library higher up in the stack. e.g. BITBOX contains primitives that sweep makes use of. But sweeping is not a primitive, so it probably doesn't belong in BITBOX.

File: lib/Util.ts [hunk 2]
```diff
@@ -49,4 +58,117 @@ export class Util {
...
+      const response: any = await axios.get(
+        `${REST_URL}rawtransactions/sendRawTransaction/${hex}`
+      )
+
+      return response.data
+    } catch (error) {
+      if (error.response && error.response.data) throw error.response.data
+      else throw error
+    }
+  }
```
Comment (christroutner) [inline lib/Util.ts]: The sweep function makes three calls to rest.bitcoin.com, which will make the unit tests clunky to mock.

File: test/integration/util.js [hunk 3]
```diff
@@ -112,4 +112,19 @@ describe(`#util`, () => {
...
+      console.log(`result: ${result}`)
+    })
+
+    it("should sweep funds", async () => {
+      const wif = "L287yGQj4DB4fbUKSV7DMHsyGQs1qh2E3EYJ21P88mXNKaFvmNWk"
+      const toAddr = "bitcoincash:qqjes5sxwneywmnzqndvs6p3l9rp55a2ug0e6e6s0a"
+      const result = await bitbox.Util.sweep(wif, toAddr)
+      console.log(`result: ${result}`)
+    })
+  })
```
Comment (christroutner) [inline test/integration/util.js]: These are just prototype tests. They exercise the basic functionality of the sweep function.

File: lib/Util.ts (hunk 1 above)
Comment (paOol) [reply lib/Util.ts]: all the `imports` should be on the top;

the `consts` should be below the `imports`;

Comment (paOol) [reply lib/Util.ts]: also try using 
`import Address from "./Address"` syntax if you are getting errors.

versus

`import { Address } from "./Address"`

File: lib/Address.ts [hunk 4]
```diff
@@ -253,6 +253,8 @@ export class Address {
     address: string | string[]
   ): Promise<AddressDetailsResult | AddressDetailsResult[]> {
     try {
+      console.log(`calling ${this.restURL}`)
```
Comment (cgcardona) [inline lib/Address.ts]: Can you remove this?

**Comment by christroutner** [comment]
Spent a few hours trying to resolve an issue with trying to import the Address.ts library into the Utils.ts library. These lines:

```
import { Address } from "./Address"
const addressLib = new Address(REST_URL)
```

Cause this error:

```
/home/trout/work/bch/bitbox-sdk/lib/Util.js:5
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
                                      ^
TypeError: Address_1.Address is not a constructor
    at Object.<anonymous> (/home/trout/work/bch/bitbox-sdk/lib/Util.js:5:39)
```

I can't figure out what's causing it. Got some help from Paul and Pete and still couldn't resolve it.

**Comment by christroutner** [comment]
Even if the above issue could be resolved, it doesn't feel right to put this high-level function into BITBOX which is a low-level library. I finished adding the features and unit tests to bch-cli-wallet in this PR:

https://github.com/Bitcoin-com/bch-cli-wallet/pull/26

**Comment by cgcardona** [comment]
I'm not quite ready to call it a day on this one.

**Pull request merged**